### ISSUE TITLE
Feature Request: add show_versions()

### ISSUE THREAD

**Issue by raybellwaves** [pytroll/satpy#1290]
See https://github.com/pydata/xarray/blob/master/xarray/util/print_versions.py#L80

Could be used in https://github.com/pytroll/satpy/blob/master/.github/ISSUE_TEMPLATE/bug_report.md

**Comment by djhoese** [comment]
Yeah I've thought about this but never got to the point of making an issue. Thanks for that. This would be a really great idea. Dependencies that we could/should include:

- [ ] Python
- [ ] numpy
- [ ] dask
- [ ] rasterio
- [ ] gdal
- [ ] PIL/pillow
- [ ] pykdtree
- [ ] numba
- [ ] scipy
- [ ] xarray

Then of course the pytroll specific ones:

- [ ] pyresample
- [ ] trollimage
- [ ] pycoast
- [ ] pydecorate
- [ ] pyspectral
- [ ] pyorbital
- [ ] aggdraw
- [ ] donfig (not currently used but will be)
- [ ] python-geotiepoints

And of course we should include the system information like xarray does.
